restapi/api/device_endpoints.go

```diff
@@ -121,6 +122,35 @@ func ResetLocation(c *gin.Context) {
 	c.JSON(http.StatusOK, GenericResponse{Message: "Device location reset"})
 }
 
+// Get the list of installed profiles
+// @Summary      get the list of profiles
+// @Description  get the list of installed profiles from the ios device
+// @Tags         general_device_specific
+// @Produce      json
+// @Success      200  {object}  map[string]interface{}
+// @Failure      500  {object}  GenericResponse
+// @Failure      404  {object}  GenericResponse
+// @Router       /device/{udid}/profiles [get]
+func GetProfiles(c *gin.Context) {
+
+	device := c.MustGet(IOS_KEY).(ios.DeviceEntry)
+
+	mcinstallconn, err := mcinstall.New(device)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, GenericResponse{Error: "mcInstall Connection not established" + err.Error()})
```

Comment:
you need to add a `return` statement after c.JSON... otherwise the function will continue to execute and just get more errors.

Reply:
please change the error to this format: 
`gin.H{"message": "Failed getting device list with error", "error": err.Error()}`

Reply:
done